File: httpcsharp/httpcsharp/models/Book.cs
namespace httpcsharp.models
{
    public class Book
    {
        public string Id { get; }
        public string AuthorId { get; set; }
        public string GenreId { get; set; }
        public string Name { get; set; }
        
        public Book(string id, string authorId, string genreId, string name)
        {
            Id = id;
            AuthorId = authorId;
            GenreId = genreId;
            Name = name;
        }

        public bool IsValid()
        {
            if (string.IsNullOrEmpty(AuthorId))
            {
                return false;
            }

            if (string.IsNullOrEmpty(GenreId))
            {
                return false;
            }

            if (string.IsNullOrEmpty(Name))
            {
                return false;
            }

            return true;
        }
    }
}

File: httpjava/src/main/java/org/stuartaroth/httpjava/models/Book.java
package org.stuartaroth.httpjava.models;

public class Book {
    private String id;
    private String authorId;
    private String genreId;
    private String name;

    public Book(String id, String authorId, String genreId, String name) {
        this.id = id;
        this.authorId = authorId;
        this.genreId = genreId;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getAuthorId() {
        return authorId;
    }

    public void setAuthorId(String authorId) {
        this.authorId = authorId;
    }

    public String getGenreId() {
        return genreId;
    }

    public void setGenreId(String genreId) {
        this.genreId = genreId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isValid() {
        if (authorId == null || authorId.equals("")) {
            return false;
        }

        if (genreId == null || genreId.equals("")) {
            return false;
        }

        if (name == null || name.equals("")) {
            return false;
        }

        return true;
    }
}
